ouroboros-network-api/src/Ouroboros/Network/PeerSelection/RelayAccessPoint.hs

```diff
@@ -53,6 +58,49 @@ data RelayAccessPoint = RelayAccessDomain  !DNS.Domain !Socket.PortNumber
                       | RelayAccessAddress !IP.IP      !Socket.PortNumber
   deriving (Eq, Ord)
 
+-- | These instances are used to serialize 'LedgerPeerSnapshot'
+-- consensus LocalStateQuery server which uses these instances
+-- for all its query responses. It appears they provide some improved
+-- debugging diagnostics over Serialize instances.
+instance ToCBOR RelayAccessPoint where
+  toCBOR = \case
+    RelayAccessDomain domain port ->
+         Codec.encodeListLen 3
+      <> Codec.encodeWord8 0
+      <> serialise' port
+      <> toCBOR domain
+    RelayAccessAddress (IP.IPv4 ipv4) port ->
+         Codec.encodeListLen 3
+      <> Codec.encodeWord8 1
+      <> serialise' port
+      <> toCBOR (IP.fromIPv4 ipv4)
+    RelayAccessAddress (IP.IPv6 ip6) port ->
+         Codec.encodeListLen 3
+      <> Codec.encodeWord8 2
+      <> serialise' port
+      <> toCBOR (IP.fromIPv6 ip6)
+    where
+      serialise' = toCBOR . toInteger
+
+instance FromCBOR RelayAccessPoint where
+  fromCBOR = do
+    listLen <- Codec.decodeListLen
+    when (listLen /= 3) . fail $    "Unrecognized RelayAccessPoint list length "
+                                 <> show listLen
+    constructorTag <- Codec.decodeWord8
+    port <- fromInteger <$> fromCBOR @Integer
+    case constructorTag of
+      0 -> do
+        domain <- fromCBOR
+        return $ RelayAccessDomain domain port
+      1 -> do
+        ip4 <- IP.IPv4 . IP.toIPv4 <$> fromCBOR
+        return $ RelayAccessAddress ip4 port
+      2 -> do
+        ip6 <- IP.IPv6 . IP.toIPv6 <$> fromCBOR
+        return $ RelayAccessAddress ip6 port
+      _ -> fail $ "Unrecognized RelayAccessPoint tag: " <> show constructorTag
```

Comment:
@bolt12 don't we have functions which do the serialisation / deserialisation of IP addresses for the peer sharing mini-protocol?

Reply:
Yes I think you are right! We can reuse them

Reply:
They depend on `NodeToNodeVersion`. Unfortunately, we don't have access to them here, so we cannot reuse them.